### DISCUSSION TITLE
Error not throwing

### DISCUSSION THREAD

**Discussion by jamalsoueidan** [edmundhung/conform#700] [Q&A]
Hi,

```js
export const customerCreateBody = zod.object({
  "customerId": zod.number(),
  "fullname": zod.string(),
  "username": zod.string().min(customerCreateBodyUsernameMin).regex(customerCreateBodyUsernameRegExp),
  "gender": zod.string(),
  "speaks": zod.array(zod.string()).min(1)
})
```

Then

```js
  const [form, {username, gender, speaks}] = useForm({
    lastResult,
    defaultValue: {
      username: '',
      speaks: ['danish'],
      gender: 'woman',
    },
    onValidate({formData}) {
      return parseWithZod(formData, {
        schema,
      });
    },
    shouldValidate: 'onBlur',
    shouldRevalidate: 'onInput',
  });

  console.log(form.dirty, form.valid, form.errors, form.value);
  ```

1. form.errors always empty, even when i change any value in any input fields while username is empty
2. Why is dirty always true when the form load? but the valid is true, it should be onLoad dirty=false and valid=true, or dirty=true and valid=false

**Comment by edmundhung** [comment]
> form.errors always empty, even when i change any value in any input fields while username is empty
I think you are looking for `form.allErrors`. `form.errors` will be presented only if you have any errors at the root, which is usually set with a `.refine()` on your zod object, e.g.

```tsx
export const customerCreateBody = z
  .object({
    "customerId": z.number(),
    "fullname": z.string(),
    "username": z.string().min(customerCreateBodyUsernameMin).regex(customerCreateBodyUsernameRegExp),
    "gender": z.string(),
    "speaks": z.array(z.string()).min(1)
  })
  .refine(data => data.fullname !== '' || data.username !== '', {
    message: 'Customer must have the fullname or username defined',
    // Optional. Zod will default it based on the position of the schema you are refining.
    // In this example, it would be `[]`, which indicate root 
    path: [],
  })
```

> Why is dirty always true when the form load? but the valid is true, it should be onLoad dirty=false and valid=true, or dirty=true and valid=false

Conform is always comparing your default value with the actual form value presented in the form. If it is considered dirty, then it's likely you have some fields forgot to rendered, or initialized with a different value for some reason. You can check the current form value by inspecting `form.value`.
